app/uk/gov/hmrc/agentassurance/connectors/DesConnector.scala

```diff
@@ -171,4 +187,27 @@ class DesConnectorImpl @Inject() (httpGet: HttpClientV2, metrics: Metrics)(impli
         result
       })
   }
+
+  private def getWithDesHeadersWithRetry[A: HttpReads](
+      apiName: String,
+      url: URL
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, x: Reads[A]): Future[A] = {
+    val desHeaderCarrier = hc.copy(
```

Comment:
we'll get duplicate headers log messages with this. We have implemented a way of adding headers depending on the environment here https://github.com/hmrc/agent-client-relationships/blob/main/app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala#L345-L357.